lib/index.js

```diff
@@ -111,130 +108,141 @@ function getPolyfillString(options) {
+	const supportedUA = ((!ua.meetsBaseline() || ua.isUnknown()) && options.unknown !== 'polyfill');
+	const taskPromise = supportedUA ? Promise.resolve({}) : getPolyfills(options)
+
+		// Build a polyfill bundle of polyfill sources sorted in dependency order
+		.then(targetedFeatures => {
+			const warnings = {unknown:[]};
+			const graph = tsort();
+
+			Object.keys(targetedFeatures).forEach(featureName => {
+				const feature = targetedFeatures[featureName];
+				const polyfill = sourceslib.getPolyfillMetaSync(featureName);
+				if (!polyfill) {
+					warnings.unknown.push(featureName);
+				} else {
+
+					graph.add(featureName);
+
+					if (polyfill.dependencies) {
+						polyfill.dependencies.forEach(depName => {
+							if (depName in targetedFeatures) {
+								graph.add(depName, featureName);
+							}
+						});
+					}
+
+					feature.detectSource = polyfill.detectSource;
+					feature.comment = featureName + ', License: ' + (polyfill.license || 'CC0') + ((feature.aliasOf && feature.aliasOf.size) ? ' (required by "' + Array.from(feature.aliasOf).join('", "') + '")' : '');
+				}
+			});
+
+			const sortedFeatures = graph.sort();
+
+			if (!options.minify) {
 				explainerComment.push(
-					'Polyfill service ' + versionStr,
-					'For detailed credits and licence information see http://github.com/financial-times/polyfill-service.',
+					'Polyfill service ' + ((process.env.NODE_ENV === 'production') ? 'v'+appVersion : 'DEVELOPMENT MODE - for live use set NODE_ENV to \'production\''),
+					'For detailed credits and licence information see https://github.com/financial-times/polyfill-service.',
 					'',
 					'UA detected: ' + uaDebugName,
 					'Features requested: ' + Object.keys(options.features),
 					''
 				);
 				if (!ua.meetsBaseline() && ua.getBaseline()) {
-					explainerComment.push('Version range for polyfill support in this family is: ' + ua.getBaseline());
-					explainerComment.push('');
+					explainerComment.push(
+						'Version range for polyfill support in ' + ua.getFamily() + ' is: ' + ua.getBaseline(),
+						''
+					);
+				}
+				explainerComment.push(...sortedFeatures
+					.filter(featureName => (targetedFeatures[featureName] && targetedFeatures[featureName].comment))
+					.map(featureName => '- ' + targetedFeatures[featureName].comment)
+				);
+				if (warnings.unknown.length) {
+					explainerComment.push(
+						'',
+						'These features were not recognised:',
+						warnings.unknown.map(s => '- '+s)
+					);
+				}
+				if ('all' in options.features) {
+					explainerComment.push('', allWarnText);
 				}
-			}
-			if ((!ua.meetsBaseline() || ua.isUnknown()) && options.unknown !== 'polyfill') {
-				return [{}, []];
 			} else {
+				explainerComment.push('Disable minification (remove `.min` from URL path) for more info');
+			}
+			output.queue('/* ' + explainerComment.join('\n * ') + ' */\n\n');
```

Comment:
The `output.queue` method can take a string, promise, stream, or a function that returns one of those.  The streamcombiner will wait for the current stream to end, then take the next thing off the queue.  If it's a function, it calls it and then looks again at the return value.  If the thing is a string, the string is transferred into the output stream and we move on to the next thing.  If it's a promise, we wait on the promise, transfer the result into the output stream and move on.  If it's a stream, we pipe it to the output, wait on the end of the stream, and then move on to the next thing.

The upshot of this is that you can queue anything, using a function wrapper to only initialise streams when we're ready to read from them, and weave those streams together with precomposed strings, which are only streamed out at the right point, and held in the queue until then.